lib/s3/managed_upload.js

```diff
@@ -370,7 +370,8 @@ AWS.S3.ManagedUpload = AWS.util.inherit({
     }
 
     if (self.partBufferLength >= self.partSize) {
-      var pbuf = Buffer.concat(self.partBuffers);
+      var pbuf = self.partBuffers.length === 1 ?
+        self.partBuffers[0] : Buffer.concat(self.partBuffers);
```

Comment:
Can you also apply this to line 389  right under `if (self.isDoneChunking && !self.isDoneSending) {` as well? 
Otherwise, looks good for us to merge in!